### PR TITLE
Update to forms, more CSS styling

### PR DESCRIPTION
**Pull request by BrettEastman**
Added box-sizing to the global css
Added a bold version of the font to the font family
Completely re-did the CSS for the shared Modal component to look decent and to allow for automatic resizing
Added working forms for both the Add a Question button and the Add Answer button
Added form components and form CSS to the shared folder in case anyone wants to use them
*Note: I'm still working on a few things, such as scrolling and adding photos, but will continue to work on them. Also, I added a window alert for the prompt if the user doesn't fill out an input, but it's not great so I will improve that.

### CODE REVIEW

File: client/source/components/questionsAnswers/LoadAddQA.jsx [hunk 1]
```diff
@@ -1,9 +1,41 @@
...
+
+  const postQuestion = (data) => {
+    const obj = {
+      body: data.question, name: data.name, email: data.email, product_id: prodID
+    };
+    return requests
+      .post('/qa/questions', obj);
+  };
+
+  const handlePost = (event, input) => {
```
Comment (aimeekang) [inline client/source/components/questionsAnswers/LoadAddQA.jsx:24]: Nice job on the post! I saw a successful 201 when I input all the required fields. 

There is some interesting behavior when I don't fill out a required field, I get the prompt, and upon closing the prompt, it still submits the form when I haven't added anything to the field that is a requirement.

File: client/source/components/questionsAnswers/LoadAddQA.jsx [hunk 2]
```diff
@@ -14,10 +46,16 @@ const LoadAddQA = function CreateLoadAddQAComponent({
         }}
       >More Answered Questions
       </button>
-      <button type="button">Add a Question</button>
+      <Modal isOpen={isOpen} onClose={onClose}>
+        <ModalFormTitle
```
Comment (aimeekang) [inline client/source/components/questionsAnswers/LoadAddQA.jsx:55]: This new ModalFormTitle looks great!

File: client/source/components/questionsAnswers/QuestionForm.jsx [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+  };
+
+  const handleFormInputChange = useCallback((event) => {
+    setFormInput({
+      ...formInput,
+      [event.target.name]: event.target.value
+    });
+  }, [formInput]);
+
+  return (
```
Comment (aimeekang) [inline client/source/components/questionsAnswers/QuestionForm.jsx:26]: Super clean - I'll likely align to what you have here :)

File: client/source/components/questionsAnswers/HelpfulQ.jsx [hunk 4]
```diff
@@ -12,6 +22,29 @@ const HelpfulQ = function CreateHelpfulQComponent({ questionHelpfulness, id, get
...
+  };
+
+  const handlePost = (event, input) => {
+    const keys = Object.keys(input);
+    for (let i = 0; i < keys.length; i += 1) {
+      if (input[keys[i]] === '') {
+        alert(`You must enter the following: ${keys[i]}`);
+      }
+    }
+    postAnswer(input)
```
Comment (aimeekang) [inline client/source/components/questionsAnswers/HelpfulQ.jsx]: I think when someone clicks out of the alert, it continues to post without the user actually updating the field that needs to have input.

File: client/source/components/shared/form/FormInput.jsx [hunk 5]
```diff
@@ -0,0 +1,43 @@
+import React from 'react';
```
Comment (aimeekang) [inline client/source/components/shared/form/FormInput.jsx:1]: Really like how you set this up for reuse!

**Review by aimeekang — APPROVED** [review]

File: client/source/components/questionsAnswers/LoadAddQA.jsx (hunk 1 above)
Comment (BrettEastman) [reply client/source/components/questionsAnswers/LoadAddQA.jsx:24]: Thank you for pointing this out and thank you for the kind words! I also saw the same post request going through despite not having all required inputs filled in. I changed the logic to account for this, so we should be good now on this.

File: client/source/components/questionsAnswers/LoadAddQA.jsx (hunk 2 above)
Comment (BrettEastman) [reply client/source/components/questionsAnswers/LoadAddQA.jsx:55]: Thank you!

File: client/source/components/questionsAnswers/QuestionForm.jsx (hunk 3 above)
Comment (BrettEastman) [reply client/source/components/questionsAnswers/QuestionForm.jsx:26]: 🙌

File: client/source/components/questionsAnswers/HelpfulQ.jsx (hunk 4 above)
Comment (BrettEastman) [reply client/source/components/questionsAnswers/HelpfulQ.jsx]: Fixed this too, so it should stop warning now after first non-updated field

File: client/source/components/shared/form/FormInput.jsx (hunk 5 above)
Comment (BrettEastman) [reply client/source/components/shared/form/FormInput.jsx:1]: Glad to help, hope you find it useful!

File: client/source/components/questionsAnswers/LoadAddQA.jsx (hunk 2 above)
Comment (BrettEastman) [reply client/source/components/questionsAnswers/LoadAddQA.jsx:55]: Also, I moved it to the shared/form folder so you can use it too!

**Pull request merged**